src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuildingSmelterCrafter.java

```diff
@@ -94,26 +94,31 @@ public int getMaxBuildingLevel()
                     final IRequest<? extends PublicCrafting> request = (IRequest<? extends PublicCrafting>) colony.getRequestManager().getRequestForToken(taskToken);
                     if (request != null)
                     {
-                        final IRecipeStorage recipeStorage = getFirstFullFillableRecipe(request.getRequest().getStack(), false);
+                        final IRecipeStorage recipeStorage = getFirstRecipe(request.getRequest().getStack());
                         if (recipeStorage != null)
                         {
                             for (final ItemStorage itemStorage : recipeStorage.getCleanedInput())
                             {
                                 int amount = itemStorage.getAmount() * request.getRequest().getCount();
-                                if (recipeOutputs.containsKey(itemStorage))
+                                if (requiredItems.containsKey(itemStorage))
                                 {
-                                    amount += recipeOutputs.get(itemStorage).getA();
+                                    amount += requiredItems.get(itemStorage).getA();
                                 }
-                                recipeOutputs.put(itemStorage, new Tuple<>(amount, false));
+                                requiredItems.put(itemStorage, new Tuple<>(amount, false));
                             }
                         }
-                    }
+                        final ItemStorage output = new ItemStorage(recipeStorage.getPrimaryOutput());
```

Comment:
seems badly alinged

Reply:
The problem is actually line 115. Fixing.